src/components/NodeEditor/NodeEditor.tsx

```diff
@@ -228,12 +319,100 @@ export default class NodeEditor extends React.PureComponent<NodeEditorProps, Nod
         );
     }
 
+    private getInitialState(): NodeEditorState {
+        const type = this.determineConfigType();
+        const config = this.context.getTypeConfig(type);
+        const show = this.props.show || false;
+        const setResultName =
+            this.props.node.router && this.props.node.router.result_name ? true : false;
+
+        return {
+            config,
+            show,
+            setResultName
+        };
+    }
+
     private getLocalizedObject(): LocalizedObject {
         if (this.props.localizations && this.props.localizations.length) {
             return this.props.localizations[0];
         }
     }
 
+    private getResultNameField(): JSX.Element {
```

Comment:
Moved this method to `NodeEditor` as both `SwitchRouter` and `GroupRouter` need a result name field